includes/REST_API/Embed_Controller.php

```diff
@@ -357,7 +357,7 @@ private function get_data_from_post( string $url ) {
 	 * @param string $url Permalink to check.
 	 * @return WP_Post|null Post object on success, null otherwise.
 	 */
-	private function url_to_post( $url ): ?WP_Post {
+	private function url_to_post( $url ): ?WP_Post { // phpcs:ignore SlevomatCodingStandard.Complexity.Cognitive.ComplexityTooHigh
```

Comment:
I do think this method should be broken up. The logic to switch site in multisite is a clean could in it's own function to me. This seems like valid and should not be ignored.

Reply:
Agreed, though it's not urgent. We have a dedicated ticket for that actually: #5065. I closed it as a maybelater because it's not really a priority.

Reply:
Solution for this one is super simple. Just move this logic to it's own method. See my POC. https://github.com/GoogleForCreators/web-stories-wp/pull/12932. Should be simply to fix up my POC and merge.